scriptworker/config.py

```diff
@@ -134,11 +114,12 @@ def check_config(config, path):
     """
     messages = []
 
-    missing_keys = set(DEFAULT_CONFIG.keys()) - set(config.keys())
+    config_copy = get_frozen_copy(config)
```

Comment:
We check `config` vs `DEFAULT_CONFIG` before we remove `credentials` from `config`, which won't work if `config` is immutable.  We could freeze it, check it, unfreeze it, remove credentials, and refreeze it, but this made more sense.  We could also leave `credentials` in there; I'm not sure if I had a strong reason to remove it other than making sure I could freeze the credentials when `freeze_values` wasn't recursive, other than being able to do a config dump without revealing secrets.

As designed.  We can revisit if there's some other way that makes more sense... which may be leaving the credentials in the config.